src/components/tabs/tabs.ts

```diff
@@ -121,17 +121,23 @@ export default class IgcTabsComponent extends EventEmitterMixin<
   private _mutationCallback({
     changes: { attributes, added, removed },
   }: MutationControllerParams<IgcTabComponent>) {
-    this.setSelectedTab(attributes.find((tab) => tab.selected));
+    const ownAttributes = attributes.filter(
+      (tab) => tab.closest(this.tagName) === this
+    );
+    const ownAdded = added.filter(({ target }) => target === this);
+    const ownRemoved = removed.filter(({ target }) => target === this);
```

Comment:
Shouldn't we check these with `target.closest(this.tagName) === this` still? Just like the attributes. In most cases these should be direct `igc-tab` children, but technically those can be wrapped in containers, which will shift the target.